tutorials/inverse/95_phantom_KIT.py

```diff
@@ -0,0 +1,166 @@
+ax.set(xlim=dip_t[[0, stop - 1]])
+
+# We know that there are 32 dipoles, so mark the first ones as well
+n_dip = 49
+assert len(peaks) % n_dip == 0  # we found them all (hopefully)
+ax.plot(dip_t[peak_idx[::n_dip]], dip_act[peak_idx[::n_dip]], "bo", zorder=4, ms=10)
+
+# Knowing we've caught the top of the first cycle of a 11 Hz sinusoid, plot onsets
+# with red X's.
+onsets = peaks - np.round(raw.info["sfreq"] / dip_freq / 4.0).astype(
+    int
+)  # shift to start
+onset_idx = onsets[onsets < stop]
+ax.plot(dip_t[onset_idx], dip_act[onset_idx], "rx", zorder=5, ms=5)
+
+# %%
+# Given the onsets are now stored in ``peaks``, we can create our events array and plot
+# on our raw data.
+
+n_rep = len(peaks) // n_dip
+events = np.zeros((len(peaks), 3), int)
+events[:, 0] = onsets + raw.first_samp
+events[:, 2] = np.tile(np.arange(1, n_dip + 1), n_rep)
+raw.plot(events=events, **raw_plot_kwargs)
+
+# %%
+# Now  we can figure out our epoching parameters and epoch the data, sanity checking
+# some values along the way knowing how the stimulation was done.
+
+# Sanity check and determine epoching params
+deltas = np.diff(events[:, 0], axis=0)
+group_deltas = deltas[n_dip - 1 :: n_dip] / raw.info["sfreq"]  # gap between 49 and 1
+assert (group_deltas > 0.8).all()
+assert (group_deltas < 0.9).all()
+others = np.delete(deltas, np.arange(n_dip - 1, len(deltas), n_dip))  # remove 49->1
+others = others / raw.info["sfreq"]
+assert (others > 0.25).all()
+assert (others < 0.3).all()
+tmax = 1 / dip_freq * 2.0  # 2 cycles
+tmin = tmax - others.min()
+assert tmin < 0
+epochs = mne.Epochs(
+    raw,
+    events,
+    tmin=tmin,
+    tmax=tmax,
+    baseline=(None, 0),
+    decim=10,
+    picks="data",
+    preload=True,
+)
+del raw
+epochs.plot(scalings=plot_scalings)
+
+# %%
+# Now we can average the epochs for each dipole, get the activation at the peak time,
+# and create an :class:`mne.EvokedArray` from the result.
+
+t_peak = 1.0 / dip_freq / 4.0
+data = np.zeros((len(epochs.ch_names), n_dip))
+for di in range(n_dip):
+    data[:, [di]] = epochs[str(di + 1)].average().crop(t_peak, t_peak).data
+evoked = mne.EvokedArray(data, epochs.info, tmin=0, comment="KIT phantom activations")
+evoked.plot_joint()
+
+# %%
+# Finally, let's fit dipoles at each dipole's peak activation time and plot the result.
+
+# sphinx_gallery_thumbnail_number = 7
+
+trans = mne.transforms.Transform("head", "mri", np.eye(4))
+sphere = mne.make_sphere_model(r0=(0.0, 0.0, 0.0), head_radius=0.08)
+cov = mne.compute_covariance(epochs, tmax=0, method="empirical")
+# We need to correct the ``dev_head_t`` because it's incorrect for these data!
+# relative to the helmet: hleft, forward, up
+translation = mne.transforms.translation(x=0.008, y=-0.025, z=-0.092)
+# pitch down (rot about x/R), roll left (rot about y/A), yaw left (rot about z/S)
+rotation = mne.transforms.rotation(
+    x=np.deg2rad(5),
+    y=np.deg2rad(-0.5),
+    z=np.deg2rad(-3),
+)
```

Comment:
I'm not too sure how to get the correct matrix - does this have to do with the marker coils and head shape digitisation? For KIT data, these are recorded in separate files (rather than being embedded in the raw MEG data). For this phantom dataset, we did marker coil measurements (_ini.mrk) but not head digitisation.

Reply:
Ahh bummer. Yes I think you need both. If you still have the phantom and know where you had the coils you could maybe redigitize then update the dataset. But that sounds like a lot of work :)

Reply:
We no longer have the phantom unfortunately, but the marker coils were attached at fixed locations (i.e. there were designated pins on the phantom), so I think it may be possible to get the coordinates for these somehow.

Reply:
Okay I think we can punt on this for now, then. We can always add these to the dataset at a later date if you track down the phantom and can do a marker measurement. Would be cool to see how close it is to my hand-optimized values. :slightly_smiling_face:

Reply:
Haha yes, would be interesting to see!